consensus/istanbul/backend/handler.go

```diff
@@ -286,3 +301,172 @@ func (sb *Backend) UnregisterPeer(peer consensus.Peer, isProxiedPeer bool) {
 		}
 	}
 }
+
+// Handshake allows the initiating peer to identify itself as a validator
+func (sb *Backend) Handshake(peer consensus.Peer) (bool, error) {
+	// Only written to if there was a non-nil error when sending or receiving
+	errCh := make(chan error)
+	isValidatorCh := make(chan bool)
+
+	sendHandshake := func() {
+		var validatorProofMsg *istanbul.Message
```

Comment:
Naming this `validatorProofMsg` is a little misleading because its actaully the `versionedEnode`